irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -115,9 +145,18 @@ void OnDemandOrderingServiceImpl::packNextProposals(
     if (it != current_proposals_.end()) {
       log_->debug("proposal found");
       if (not it->second.empty()) {
-        proposal_map_.emplace(round, emitProposal(round));
-        log_->debug("packNextProposal: data has been fetched for {}", round);
-        round_queue_.push(round);
+        log_->debug("Mutable proposal generation for round {}", round);
+        auto txs = getTransactions(transaction_limit_, it->second);
+        if (not txs.empty()) {
+          log_->debug("Number of transactions in proposal = {}", txs.size());
+          auto proposal = proposal_factory_->unsafeCreateProposal(
+              round.block_round,
+              iroha::time::now(),
+              std::move(txs) | boost::adaptors::indirected);
+          proposal_map_.emplace(round, std::move(proposal));
+          log_->debug("packNextProposal: data has been fetched for {}", round);
+          round_queue_.push(round);
+        }
       }
       current_proposals_.erase(it);
```

Comment:
Information about lost transactions should be logged, probably by checking `it->second.unsafe_size()`.